src/app/project/project-details-tab/project-details-tab.component.ts

```diff
@@ -25,69 +22,120 @@ declare let L;
 })
 export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDestroy {
 	@ViewChild('map') private mapContainer: ElementRef;
+
+  private ngbModal: NgbModalRef = null;
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public pathAPI: string;
   public project;
+  public activities;
   public loading = true;
   public commentPeriod = null;
-  public map: L.Map = null;
-  public appFG = L.featureGroup(); // group of layers for subject app
   public multipleExistingPlans: boolean;
   public overlappingDistrictsListString: string;
-  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
-  readonly defaultBounds = L.latLngBounds([48, -139], [61, -114]); // all of BC
+
+  public map: Map = null;
+  public appFG: FeatureGroup = null; // group of layers for subject app
+  public defaultBounds: LatLngBounds = null; // all of BC
 	readonly defaultBoundsObject = {southWest: {lat: 48, lng: -139}, northEast: {lat: 61, lng: -114}}; // Converted for other parsing
 	public bounds = {southWest: {lat: null, lng: null}, northEast: {lat: null, lng: null}}; // Bounds object for keeping track of bounds
-  private ngbModal: NgbModalRef = null;
   public shapefiles: Document[] = [];
 	public convertedShapefiles = [];
-  public pathAPI: string;
+  private L: typeof import('leaflet');
 
   constructor(
-    private storageService: StorageService,
     private elementRef: ElementRef,
     public configService: ConfigService,
     private route: ActivatedRoute,
     private utils: Utils,
+    private scriptLoader: ScriptLoaderService,
+    private projectService: ProjectService,
   ) { }
 
   ngOnInit() {
+    // Remove any existing map
     if (this.map) {
       this.map.remove();
       this.map = undefined;
     }
-    this.project = this.storageService.state.currentProject.data;
-    this.multipleExistingPlans = Array.isArray(this.project.existingLandUsePlans);
-    this.overlappingDistrictsListString = this.stringifyOverlappingDistricts(this.project.overlappingRegionalDistricts);
-    this.commentPeriod = this.project.commentPeriodForBanner;
-    this.route.data.subscribe((res: any) => {
-      if (Array.isArray(this.project?.shapefiles) && this.project?.shapefiles.length > 0) {
-        return;
+
+    const remoteApiPath = window.localStorage.getItem('from_public_server--remote_api_base_path');
+    this.pathAPI = _.isEmpty(remoteApiPath) ? 'http://localhost:3000/api' : remoteApiPath;
+
+    // Try to get project from parent resolver
+    const parentWithProject = this.route.pathFromRoot.find(r => r.snapshot.data?.projectAndBanner);
```

Comment:
If you're just getting data from the parent resolver, I think it would be clearer to write:

```suggestion
    const projectFromParentResolver = this.route.pathFromRoot.find(r => r.snapshot.data?.projectAndBanner);
```

Reply:
Technically it's the project and the banner, so it would be inaccurate to say it's just the project. I'll still rename it to make it more clear though.